src/core/qgsnetworkaccessmanager.cpp

```diff
@@ -275,6 +293,58 @@ void QgsNetworkAccessManager::onReplyDownloadProgress( qint64 bytesReceived, qin
   }
 }
 
+#ifndef QT_NO_SSL
+void QgsNetworkAccessManager::onReplySslErrors( const QList<QSslError> &errors )
+{
+  QNetworkReply *reply = qobject_cast< QNetworkReply *>( sender() );
+  Q_ASSERT( reply );
+  Q_ASSERT( reply->manager() == this );
+
+  QTimer *timer = reply->findChild<QTimer *>( QStringLiteral( "timeoutTimer" ) );
+  if ( timer && timer->isActive() )
+  {
+    QgsDebugMsg( QStringLiteral( "Stopping network reply timeout whilst SSL error is handled" ) );
+    timer->stop();
+  }
+  bool ok = false;
+  int requestId = reply->property( "requestId" ).toInt( &ok );
+  if ( ok )
+    emit requestEncounteredSslErrors( requestId, errors );
+
+  emit sslErrorsOccurred( reply, errors );
```

Comment:
maybe worth a comment that in main thread this will trigger ssl handler's code and return once the errors are handled, while in worker thread the signal will be queued (and return immediately)